src/pre_commit_terraform/_run_on_whole_repo.py

```diff
@@ -0,0 +1,102 @@
+"""Common functions to check if a hook is run on the whole repository."""
+
+import logging
+import re
+import subprocess
+from importlib.resources import files as access_artifacts_of
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def is_function_defined(func_name: str, scope: dict) -> bool:
+    """
+    Check if a function is defined in the global scope.
+
+    Args:
+        scope (dict): The scope (usually globals()) to check in.
```

Comment:
If you're ever going to integrate Sphinx, beware that it's expected that you use RST syntax in docstrings:
```suggestion
        scope (dict): The scope (usually ``globals()``) to check in.
```
(this is a simplistic example that would probably be `` :py:func:`globals` `` IRL)